### PR TITLE
37 password managing

### LINKED ISSUE #37: Password managing
To enhance password managing(same password for everything), there needs to be a system to control and check the password.

Where are passwords used / could / should be used:

- [ ] ssh login/passwd
- [ ] website login (nginx config)
- [ ] wifi hotspot

Where are passwords generated / changed:
- [ ] passwd
- [ ] (WIP: provisioning system, uses `sudo passwd -u mirte ... `)

TODO:
- [ ] Where to store the password and how secure should it be?
- - Probably not too secure, plaintext file should be fine?
- [ ] build login to nginx
- - [ ] do not check when on .42.x/43.x network
- - [ ] also vscode licence terms check page
- - [ ] Overview page of enabled webpages
- [ ] build PAM module to catch passwd password changes
- [ ] change wifi password on change(after reboot?)
- [ ] add password change page?

### PR DESCRIPTION
**Pull request by ArendJan**
Fixes #37 

# Components:

## Pam:
This Pluggable Authentication Module is hooked into the passwd calls. Before entering a new password it will show the user a warning that it stores the password and that it should not be a safe password. After entering, it will show the password and store it in `/home/mirte/.wifi_pwd` for the wifi hotspot functionality.

## ~~password_manager:~~
~~This python script will watch `/usr/local/src/mirte/mirte-install-scripts/config/pam/users.json` and `/usr/local/src/mirte/mirte-install-scripts/config/web/newpasswd` for changes. When there are changes, it updates:~~

### Login:
when a user is connecting to any webpage and is not logged in and is not from a 'secure' source (192.168.42/43/137.x), then the user gets a http basic auth popup. When login fails, a custom error page is show how to login. Nginx uses pam to check the username & password.

### Code:
The VsCode web server uses a different port from the other pages, as it does not like being in a 'directory'. Nginx will still check for the password. It is possible to go to http://192.168.42.1/code/ and you'll get redirected to the correct port.

### CODE REVIEW

**Comment by ArendJan** [comment]
After our discussion:
- Remove the option to change password from web
- Remove python script password_manager
- Change nginx to use pam
- Let pam module do all the settings (extra bash script)
- (optional) let pam module ask user to change wifi password

File: create_user.sh [hunk 1]
```diff
@@ -2,13 +2,15 @@
...
 # Add mirte user with sudo rights
 #TODO: user without homedir (create homedir for user)
 sudo useradd -m -G sudo,audio -s /bin/bash mirte
 sudo mkdir /home/mirte/workdir
 sudo chown mirte:mirte /home/mirte/workdir
-echo -e "mirte_mirte\nmirte_mirte" | sudo passwd mirte
+
+sudo ./install_pam.sh # setup pam before changing the password for the mirte user.
+
+echo "mirte:{new_password}" | sudo chpasswd
```
Comment (mklomp) [inline create_user.sh:13]: Where is {new_password} set?

File: nginx.conf
Comment (mklomp) [inline nginx.conf:1]: There is a lot of duplicate code here. It seems nginx can also have shared [common configs](https://serverfault.com/questions/373578/how-can-i-configure-nginx-locations-to-share-common-configuration-options. I think that cleans it up a lot.)

File: pam/image.png
Comment (mklomp) [inline pam/image.png:1]: This is is now different, right?

**Review by mklomp — APPROVED** [review]
Perfect. I think especially the nginx.conf can be cleaned up a bit, but looks great.

File: create_user.sh (hunk 1 above)
Comment (ArendJan) [reply create_user.sh:13]: nowhere, probably some find replace mistake

File: nginx.conf
Comment (ArendJan) [reply nginx.conf:1]: I searched for "nginx macros", but didn't find one, this is way better. I've implemented it and resolved one warning!

File: pam/image.png
Comment (ArendJan) [reply pam/image.png:1]: jup, that image was old, the current file is okay. Shows that only for the mirte user it shows any output

**Pull request merged**